pkg/kubelet/apis/kubeletconfig/validation/validation.go

```diff
@@ -100,5 +100,14 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration) error
 				val, kubetypes.NodeAllocatableEnforcementKey, kubetypes.SystemReservedEnforcementKey, kubetypes.KubeReservedEnforcementKey))
 		}
 	}
+	switch kc.HairpinMode {
+	case "":
```

Comment:
I don't think `""` is a valid value. Defaulting should happen before conversion to the internal type before validation, in which case if it was `""` it would be defaulted to `kubeletconfig.PromiscuousBridge`. See https://github.com/kubernetes/kubernetes/blob/f5f6f3e715cb8dfbd9657a4229c77ec6a5eab135/pkg/kubelet/apis/kubeletconfig/v1alpha1/defaults.go#L189

Reply:
>  Defaulting should happen before conversion to the internal type before validation

Agree. You are right! It's fixed now. PTAL, Thanks!